### PR TITLE
Updated dependencies & gradle

### PR DESCRIPTION
**Pull request by p-louis**
* DEL nexus dependencies as they are not compatible with gradle 7 
* ADD publishing information moved to native gradle functions
* CHG Update ktor to 2.1.3
* CHG Update kotlin to 1.7.21
* CHG Update java-saml to 2.9.0

### CODE REVIEW

File: build.gradle.kts [hunk 1]
```diff
@@ -47,8 +40,38 @@ tasks.withType<KotlinCompile> {
 // ability to publish artifact to maven local
```
Comment (weickmanna) [inline build.gradle.kts]: Comment is no longer correct. This is for publishing to Maven Central

File: build.gradle.kts [hunk 2]
```diff
@@ -47,8 +40,38 @@ tasks.withType<KotlinCompile> {
...
+                    }
+                }
+                developers {
+                    developer {
+                        name.set("Alexander Weickmann")
+                        email.set("alexander.weickmann@gmail.com")
+                        url.set("https://github.com/weickmanna")
+                        organization.set("linked-planet GmbH")
+                        organizationUrl.set("https://linked-planet.com")
+                    }
```
Comment (weickmanna) [inline build.gradle.kts:63]: Feel free to add yourself as a developer of course ;)

File: src/main/kotlin/com/linkedplanet/ktor/Saml.kt [hunk 3]
```diff
@@ -59,15 +53,13 @@ fun ApplicationCall.getServletRequest(): HttpServletRequest {
...
     val servletApplicationResponse = getAsyncServletApplicationCall().response
     val responseField = ServletApplicationResponse::class.java.getDeclaredField("servletResponse")
     responseField.isAccessible = true
     return responseField.get(servletApplicationResponse) as HttpServletResponse
 }
 
-@OptIn(EngineAPI::class)
 private fun ApplicationCall.getAsyncServletApplicationCall(): AsyncServletApplicationCall {
     val routingApplicationCall = (request.call as RoutingApplicationCall)
     val callField = RoutingApplicationCall::class.java.getDeclaredField("call")
```
Comment (weickmanna) [inline src/main/kotlin/com/linkedplanet/ktor/Saml.kt:65]: Does this still work in the new version of ktor?

**Review by weickmanna — CHANGES_REQUESTED** [review]

File: src/main/kotlin/com/linkedplanet/ktor/Saml.kt (hunk 3 above)
Comment (p-louis) [reply src/main/kotlin/com/linkedplanet/ktor/Saml.kt:65]: No clue, but at least the compiler is not complaining

Comment (p-louis) [reply src/main/kotlin/com/linkedplanet/ktor/Saml.kt:65]: Alright I've checked it with ktor 2.1.3 and this method still works with the `call` field on the `RoutingApplicationCall` containing a `JettyApplicationCall` (given jetty as engine), which extends the `AsyncServletApplicationCall`

**Review by weickmanna — APPROVED** [review]

**Pull request merged**